File: WebFt30/src/app/core/store/models/monitoring-alarm-event.ts
import { MonitoringAlarmType } from './monitoring-alarm-type';
import { MonitoringChange } from './monitoring-result';

export enum MonitoringAlarmLevel {
  Minor,
  Major,
  Critical
}

export enum MonitoringAlarmStatus {
  Active,
  Resolved
}

export class MonitoringAlarmEvent {
  id!: number;
  monitoringAlarmGroupId!: number;
  monitoringPortId!: number;
  monitoringAlarmId!: number;
  monitoringResultId!: number;
  type!: MonitoringAlarmType;
  distanceMeters!: number | null;
  at!: Date;

  oldLevel!: MonitoringAlarmLevel | null;
  level!: MonitoringAlarmLevel;

  oldStatus!: MonitoringAlarmStatus | null;
  status!: MonitoringAlarmStatus;

  change!: MonitoringChange | null;
}

File: Backend/src/Fibertest30.Application/Alarm/Models/MonitoringAlarmEvent.cs
namespace Fibertest30.Application;

public class MonitoringAlarmEvent : INotificationEvent
{
    public int Id { get; init; }
    public int MonitoringAlarmId { get; set; }
    public int MonitoringAlarmGroupId { get; init; }
    public int MonitoringPortId { get; set; }
    public int MonitoringResultId { get; set; }
    public MonitoringAlarmType Type { get; set; }
    public double? DistanceMeters { get; set; }
    public DateTime At { get; set; }
    public MonitoringAlarmStatus? OldStatus { get; set; }
    public MonitoringAlarmStatus Status { get; set; }
    public MonitoringAlarmLevel? OldLevel { get; set; }
    public MonitoringAlarmLevel Level { get; set; }
    public MonitoringChange? Change { get; init; }



}